src/js/references/hooks.ts

```diff
@@ -128,3 +130,55 @@ export function useUpdateSourceTypes(
         register,
     };
 }
+
+/**
+ * Check if the logged in account has the passed `right` on the reference detail is loaded for.
+ *
+ * @param reference - The reference to check
+ * @param right - The right to check for (eg. modify_otu)
+ * @returns Whether the right is possessed by the account
+ */
+export function useCheckReferenceRight(reference: Reference, right: string) {
+    const { data: account, isLoading } = useFetchAccount();
+
+    if (isLoading || reference === null) {
+        return;
+    }
+
+    if (account.administrator_role === AdministratorRoles.FULL) {
+        return true;
+    }
+
+    const user = find(reference.users, { id: account.id });
+
+    if (user && user[right]) {
+        return true;
+    }
+
+    // Groups user is a member of.
+    const memberGroups = account.groups;
+
+    // Groups in common between the user and the registered ref groups.
+    const groups = filter(reference.groups, group => includes(memberGroups, group.id));
+
+    if (!groups) {
+        return false;
+    }
+
+    return some(groups, { [right]: true });
+}
+
+/**
+ * Get a boolean indicating whether the logged in account can modify the OTUs of the
+ * reference detail is loaded for.
```

Comment:
```suggestion
 * Get a boolean indicating whether the current user can modify the OTUs of a reference.
```
Careful on the wording, any reference id can be passed here